modules/@apostrophecms/migration/lib/addMissingSchemaFields.js

```diff
@@ -0,0 +1,141 @@
+const _ = require('lodash');
+const { klona } = require('klona');
+
+module.exports = (self) => {
+  return {
+    async addMissingSchemaFields() {
+      let scans = 0; let updates = 0;
+      const lastPropLists = await self.getLastPropLists();
+      const propLists = self.getPropLists();
+      let changesToPropLists = false;
+      for (const name of Object.keys(propLists)) {
+        if (!_.isEqual(lastPropLists?.[name], propLists[name])) {
+          changesToPropLists = true;
+          scans++;
+          updates += await self.addMissingSchemaFieldsForDocType(name, propLists[name]);
```

Comment:
This method accepts only 1 parameter below

Reply:
The migration phase **will run** on any startup. **Including** an empty task. So omitting the call from the task is just eliminating redundancy.

At a future time we might reintroduce an optional behavior where the migration does not run in production except in the task, but that behavior has effectively not been possible before, so I don't have to introduce that feature to fix the redundancy bug.